scripts/zkai_native_d128_compressed_outer_statement_binary_accounting_gate.py

```
@@ -0,0 +1,659 @@
+        raise BinaryTypedProofAccountingGateError("CLI upstream serialization status drift")
+    if summary["proof_payload_kind"] != PROOF_PAYLOAD_KIND:
+        raise BinaryTypedProofAccountingGateError("CLI proof payload kind drift")
+    accounting_base.validate_safety(summary["safety"])
+    accounting_base.validate_size_constants(summary["size_constants"])
+    rows = summary.get("rows")
+    if not isinstance(rows, list) or len(rows) != 1:
+        raise BinaryTypedProofAccountingGateError("CLI row count drift")
+    validate_cli_row(rows[0], EXPECTED_ROLE)
+
+
+def validate_cli_row(row: Any, expected: dict[str, Any]) -> None:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `row`

(ANN401)